### ISSUE TITLE
Rule Change: Suggestion for `require-await` to remove `async` keyword

### ISSUE THREAD

**Issue by reduckted** [eslint/eslint#18713]
### What rule do you want to change?

require-await

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const arrow = async () => {
    return;
};
```

### What does the rule currently do for this code?

No suggestions are provided.

### What will the rule do after it's changed?

A suggestion will be provided to remove the `async` keyword.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Related request in `@typescript-eslint`: https://github.com/typescript-eslint/typescript-eslint/issues/9621

**Comment by Tanujkanti4441** [comment]
Hi @reduckted, thanks for the issue,

i think it make sense to add such suggestion to `require-await` rule for removing `async` keyword as it is also mentioned in the docs too that 
>Asynchronous functions that don’t use await might not need to be asynchronous functions and could be the unintentional result of refactoring.

@eslint/eslint-team should we go for this?

**Comment by fasttime** [comment]
Sounds good 👍

**Comment by reduckted** [comment]
Yep, I've got the changes ready to go. I'll make a pull request shortly. 🚀
